### PR TITLE
Mark LogOff-triggered disconnections as user-initiated

### PR DESCRIPTION
**Pull request by Netshroud**
Had to do a few hacky things I don't particularly like. Any suggestions to improve this?

Ideally, this probably needs a full-blown state machine.

### CODE REVIEW

**Comment by Netshroud** [comment]
Re-did this a lot simpler to resolve #181.

File: SteamKit2/SteamKit2/Steam/CMClient.cs [hunk 1]
```diff
@@ -83,6 +83,13 @@ public IPAddress LocalIP
         /// </value>
         public TimeSpan ConnectionTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets whether or not this <see cref="CMClient"/> should imminently expect the server to close the connection.
+        /// If this is true when the connection is closed, the <see cref="SteamClient.DisconnectedCallback"/>'s <see cref="SteamClient.DisconnectedCallback.UserInitiated"/> property
+        /// will be set to <c>true</c>.
+        /// </summary>
+        public bool ExpectDisconnection { get; set; }
```
Comment (VoiDeD) [inline SteamKit2/SteamKit2/Steam/CMClient.cs]: Don't think this needs to be entirely public, consumers might be able to misuse it.

Comment (Netshroud) [inline SteamKit2/SteamKit2/Steam/CMClient.cs]: I was thinking that, but I don't want our own `ClientMsgHandlers` to be able to do things that consumer-built ones cannot.

Comment (VoiDeD) [inline SteamKit2/SteamKit2/Steam/CMClient.cs]: CMClient could likely handle that logic based on the emsg being passed to `Send`.

**Pull request closed without merging**